src/attributes/utils/data.ts

```diff
@@ -208,31 +216,102 @@ export const getReferenceAttributeDisplayData = (
     ...attribute.data,
     references:
       referencePages?.length > 0 && attribute.value?.length > 0
-        ? attribute.value.map(value => {
-            const reference = referencePages.find(
-              reference => reference.id === value
-            );
-            return {
-              label: reference.title,
-              value: reference.id
-            };
-          })
+        ? mapPagesToChoices(
+            attribute.value.map(value => {
+              const reference = referencePages.find(
+                reference => reference.id === value
+              );
+              return { ...reference };
```

Comment:
Am I missing something or it is the same as `return reference`?

Reply:
It must return object, while reference might not be found at all.